daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/api/DFDLParserUnparser.scala

```diff
@@ -236,8 +236,12 @@ object DFDL {
    * Thrown by the DaffodilUnparseConentHandler when an unexpected error
    * occurs, this usually represents a bug in Daffodil
    */
-  class DaffodilUnhandledSAXException(description: String, cause: Exception)
-    extends SAXException(description, cause)
+  class DaffodilUnhandledSAXException(description: String, cause: Throwable)
+    extends SAXException(description, new Exception(cause)) {
```

Comment:
Why the Exception wrap? Is that because SAXException will only accept an Exception as the cause argument? 

I would suggest that unless we need our constructors to accept Throwable, we should also accept only Exception and then we wouldn't need to wrap Exception to satisfy SAXException's constructor.

Reply:
Correct, SAXException accepts only an Exception as its cause argument.  Agreed, DaffodilUnhandledSAXException should accept only Exception and refrain from wrapping the cause twice.